llvm/lib/Target/AIE/AIETargetMachine.cpp

```diff
@@ -261,6 +275,26 @@ void AIEBaseTargetMachine::registerDefaultAliasAnalyses(AAManager &AAM) {
   if (EnableCustomAliasAnalysis)
     AAM.registerFunctionAnalysis<AIEBaseAA>();
 }
+
+/// Predicate for Internalize pass.
+/// Preserve functions for names passed via FunctionSkipList. These are usually
+/// entry points which shouldn't be internalized to let linker know of them
+/// (default entry point is 'main'). Preserve function declarations, as if we
+/// internalize a declaration, it would not be accessible outside of its
+/// translation unit, which could lead to linker errors if the function is
+/// used/defined elsewhere.
```

Comment:
I think the comment is a bit convoluted, maybe simply write `Preserve functions that can be an entry point or that have uses within the Module`?

Reply:
done